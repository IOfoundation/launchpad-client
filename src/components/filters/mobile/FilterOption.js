import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import ArrowRight from '../../shared/ArrowRight';

class FilterOption extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      selectedfilterOption: null,
    };
  }
  uncheckAll() {
    const checks = document.querySelectorAll(
      '.dropdown-input-container' + ' input[type="checkbox"]'
    );
    for (const i = 0; i < checks.length; i++) {
      const check = checks[i];
      if (!check.disabled) {
        check.checked = false;
      }
    }
  }
  handleClick = selectedfilterOption => () => {
    const {name} = selectedfilterOption;
    this.uncheckAll();
    return this.props.handleOnChangeFilterOptions('category', name);
  };
  _toggleSubOption(event, selectedfilterOption) {
    this.props.handleOnChangeFilterOptions(
      'category',
      selectedfilterOption.name
    );
    this.uncheckAll();
    if (selectedfilterOption.children.length > 0) {
      this.setState({selectedfilterOption, subDropdownOpen: true});
    }
  }
  _renderSubOptions(filterOption) {
    return (
      <ul>
        {filterOption.children.map(child => (
          <li key={child.id} onClick={this.handleClick(child)}>
            {child.name}
          </li>
        ))}
      </ul>
    );
  }
  _renderOptions() {
    return (
      <ul>
        {this.props.filterOptions.map(filterOption => (
          <li key={filterOption.id}>
            <input type="checkbox" id={filterOption.name} />
            <div className="hover">
              <a onClick={e => this._toggleSubOption(e, filterOption)}>
                {filterOption.name}
              </a>
              {filterOption.children.length > 0 && (
                <label htmlFor={filterOption.name}>
                  <ArrowRight
                    className="second-level-icon"
                    size={20}
                    style={{color: '#fff', verticalAlign: 'middle'}}
                  />
                </label>
              )}
            </div>
            {filterOption.children.length > 0 && (
              <div className="filters-third-level">
                {this._renderSubOptions(filterOption)}
              </div>
            )}
          </li>
        ))}
      </ul>
    );
  }
  render() {
    const {filterName} = this.props;
    return (
      <ul className="filter-list">
        <li>
          <input type="checkbox" id={filterName} />
          <div className="select">
            <label>{filterName}</label>
            <label htmlFor={filterName}>
              <ArrowRight
                className="first-level-icon"
                size={20}
                style={{color: '#fff', verticalAlign: 'middle'}}
              />
            </label>
          </div>
          <div className="filters-second-level">{this._renderOptions()}</div>
        </li>
      </ul>
    );
  }
}

FilterOption.propTypes = {
  filterName: PropTypes.string.isRequired,
  filterOptions: PropTypes.array.isRequired,
  handleOnChangeFilterOptions: PropTypes.func.isRequired,
};
export default FilterOption;
