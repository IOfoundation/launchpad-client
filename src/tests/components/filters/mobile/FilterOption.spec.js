import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FilterOption from 'components/filters/mobile/FilterOption';
import filtersFixture from 'tests/fixtures/filters';

const createProps = props => {
  return {
    filterName: props.filterName,
    filterOptions: props.filterOptions,
    handleOnChangeFilterOptions: jest.fn(),
  };
};

xdescribe('<FilterByText />', () => {
  it('renders snapshot for mobile FilterOption component', () => {
    const props = createProps({
      filterName: 'Business Services',
      filterOptions: filtersFixture.businessServices,
    });
    const wrapper = shallow(<FilterOption {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
