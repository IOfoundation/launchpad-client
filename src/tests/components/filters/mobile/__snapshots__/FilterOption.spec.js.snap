// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterByText /> renders snapshot for mobile FilterOption component 1`] = `
<ul
  className="filter-list"
>
  <li>
    <input
      id="Business Services"
      type="checkbox"
    />
    <div
      className="select"
    >
      <label>
        Business Services
      </label>
      <label
        htmlFor="Business Services"
      >
        <ArrowRight
          className="first-level-icon"
          size={20}
          style={
            Object {
              "color": "#fff",
              "verticalAlign": "middle",
            }
          }
        />
      </label>
    </div>
    <div
      className="filters-second-level"
    >
      <ul>
        <li
          key="2"
        >
          <input
            id="Financial Management"
            type="checkbox"
          />
          <div
            className="hover"
          >
            <a
              onClick={[Function]}
            >
              Financial Management
            </a>
            <label
              htmlFor="Financial Management"
            >
              <ArrowRight
                className="second-level-icon"
                size={20}
                style={
                  Object {
                    "color": "#fff",
                    "verticalAlign": "middle",
                  }
                }
              />
            </label>
          </div>
          <div
            className="filters-third-level"
          >
            <ul>
              <li
                key="3"
                onClick={[Function]}
              >
                Bookkeeping/Accounting Software Training
              </li>
              <li
                key="4"
                onClick={[Function]}
              >
                Budget, Financial Statement & Cash Flow Training
              </li>
            </ul>
          </div>
        </li>
        <li
          key="10"
        >
          <input
            id="Capital"
            type="checkbox"
          />
          <div
            className="hover"
          >
            <a
              onClick={[Function]}
            >
              Capital
            </a>
            <label
              htmlFor="Capital"
            >
              <ArrowRight
                className="second-level-icon"
                size={20}
                style={
                  Object {
                    "color": "#fff",
                    "verticalAlign": "middle",
                  }
                }
              />
            </label>
          </div>
          <div
            className="filters-third-level"
          >
            <ul>
              <li
                key="11"
                onClick={[Function]}
              >
                Accelerator Programs
              </li>
              <li
                key="12"
                onClick={[Function]}
              >
                Equity Investors: Angels
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </li>
</ul>
`;
